### PR TITLE
Add sentence-transformers embeddings

### PR DESCRIPTION
**Pull request by jonbesga**

### CODE REVIEW

**Comment by jonbesga** [comment]
> Would this change the `sentence-transformers` to be a regular dependency rather than a dev-dependency? Or should we add it to the `extras`? My concern is package size / installation time.

Is not. You need to install it separately precisely because I wanted to avoid the size issue

File: src/neo4j_genai/embeddings/sentence_transformers.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        except ImportError:
+            raise ImportError(
+                "Could not import sentence_transformers python package. "
+                "Please install it with `pip install sentence-transformers`."
+            )
+
+        self.model = SentenceTransformer(model)
+
+    def embed_query(self, text: str) -> list[float]:
+        return self.model.encode([text])
```
[the remark this answers is not in the document]
Comment (jonbesga) [reply src/neo4j_genai/embeddings/sentence_transformers.py:17]: Fair point. I'll add a test to cover this case, and yes. Is returning a float32 Numpy array

[the remark this answers is not in the document]
Comment (willtai) [reply src/neo4j_genai/embeddings/sentence_transformers.py:17]: It returns a 2d `numpy.array` ([See docs](https://sbert.net/examples/applications/computing-embeddings/README.html?highlight=sentencetransformer#sentence_transformers.SentenceTransformer))

File: src/neo4j_genai/embeddings/sentence_transformers.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def __init__(self, model="all-MiniLM-L6-v2"):
+        try:
+            from sentence_transformers import SentenceTransformer
+        except ImportError:
+            raise ImportError(
+                "Could not import sentence_transformers python package. "
+                "Please install it with `pip install sentence-transformers`."
+            )
+
+        self.model = SentenceTransformer(model)
```
Comment (willtai) [inline src/neo4j_genai/embeddings/sentence_transformers.py:14]: Will the user be interested in other arguments to SentenceTransformer ([see reference](https://sbert.net/examples/applications/computing-embeddings/README.html?highlight=sentencetransformer#sentence_transformers.SentenceTransformer)). If so, maybe we should use **args, **kwargs?

File: src/neo4j_genai/embeddings/sentence_transformers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonbesga) [reply src/neo4j_genai/embeddings/sentence_transformers.py]: e63b7fdf4745245cb0954474364d980e3badc7bd

File: src/neo4j_genai/embeddings/sentence_transformers.py (hunk 2 above)
Comment (jonbesga) [reply src/neo4j_genai/embeddings/sentence_transformers.py]: e63b7fdf4745245cb0954474364d980e3badc7bd

**Review by stellasia — APPROVED** [review]

**Pull request merged**